src/app/clusters/localization-configuration-server/localization-configuration-cluster.cpp

```diff
@@ -0,0 +1,180 @@
+
+CHIP_ERROR LocalizationConfigurationCluster::Startup(ServerClusterContext & context)
+{
+    ReturnErrorOnFailure(DefaultServerCluster::Startup(context));
+    AttributePersistence persistence(context.attributeStorage);
+
+    // Load the active locale from persistence
+    chip::app::Storage::String<kActiveLocaleMaxLength> storedLocale;
+    if (persistence.LoadString({ kRootEndpointId, LocalizationConfiguration::Id, Attributes::ActiveLocale::Id }, storedLocale))
+    {
+        SetActiveLocale(storedLocale.Content());
+    }
+    else
+    {
+        ChipLogProgress(AppServer, "Failed to load active locale from persistence, using default locale");
+        ReturnErrorOnFailure(persistence.StoreString(
+            { kRootEndpointId, LocalizationConfiguration::Id, Attributes::ActiveLocale::Id }, mActiveLocale));
+    }
+
+    return CHIP_NO_ERROR;
+}
+
+DataModel::ActionReturnStatus LocalizationConfigurationCluster::ReadAttribute(const DataModel::ReadAttributeRequest & request,
+                                                                              AttributeValueEncoder & aEncoder)
+{
+    switch (request.path.mAttributeId)
+    {
+    case FeatureMap::Id:
+        // No features defined for the cluster.
+        return aEncoder.Encode<uint32_t>(0);
+    case ClusterRevision::Id:
+        return aEncoder.Encode(LocalizationConfiguration::kRevision);
+    case ActiveLocale::Id:
+        return aEncoder.Encode(GetActiveLocale());
+    case SupportedLocales::Id:
+        return ReadSupportedLocales(aEncoder);
+    }
+    return Status::UnsupportedAttribute;
+}
+
+DataModel::ActionReturnStatus LocalizationConfigurationCluster::WriteAttribute(const DataModel::WriteAttributeRequest & request,
+                                                                               AttributeValueDecoder & aDecoder)
+{
+    AttributePersistence persistence(mContext->attributeStorage);
+    switch (request.path.mAttributeId)
+    {
+    case ActiveLocale::Id: {
+        CharSpan label;
+        ReturnErrorOnFailure(aDecoder.Decode(label));
+        ChipLogProgress(AppServer, "Setting active locale to %s", label.data());
+        // Store the string in persistence
+        Storage::String<kActiveLocaleMaxLength> shortString;
+        VerifyOrReturnError(shortString.SetContent(label), Status::Failure);
+        ReturnValueOnFailure(persistence.StoreString(request.path, shortString), Status::Failure);
+
+        return SetActiveLocale(label);
+    }
+    }
+    return Status::UnsupportedAttribute;
+}
+
+CHIP_ERROR LocalizationConfigurationCluster::Attributes(const ConcreteClusterPath & path,
+                                                        ReadOnlyBufferBuilder<DataModel::AttributeEntry> & builder)
+{
+    AttributeListBuilder attributeListBuilder(builder);
+    return attributeListBuilder.Append(Span(kMandatoryMetadata), {}, {});
+}
+
+Protocols::InteractionModel::Status LocalizationConfigurationCluster::SetActiveLocale(CharSpan activeLocale)
```

Comment:
lets return DataModel::ActionReturnStatus and make this handle serialisation as well.